clients/rust/tests/deactivate_stake.rs

```diff
@@ -0,0 +1,493 @@
+        rent_epoch: account.rent_epoch,
+    };
+
+    context.set_account(&stake_pda, &updated_stake.into());
+
+    // When we try to deactivate with an invalid authority.
+
+    let fake_authority = Keypair::new();
+
+    let deactivate_ix = DeactivateStakeBuilder::new()
+        .config(config)
+        .stake(stake_pda)
+        .stake_authority(fake_authority.pubkey())
+        .amount(50)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[deactivate_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &fake_authority],
+        context.last_blockhash,
+    );
+    let err = context
+        .banks_client
+        .process_transaction(tx)
+        .await
+        .unwrap_err();
+
+    // Then we expect an error.
+
+    assert_custom_error!(err, StakeError::InvalidAuthority);
+}
+
+#[tokio::test]
+async fn fail_deactivate_stake_with_zero_amount() {
+    let mut context = ProgramTest::new("stake_program", paladin_stake::ID, None)
+        .start_with_context()
+        .await;
+
+    // Given a config account and a validator's vote account.
+
+    let config = create_config(&mut context).await;
+    let validator = Pubkey::new_unique();
+    let authority = Keypair::new();
+    let vote = create_vote_account(&mut context, &validator, &authority.pubkey()).await;
+
+    // And a stake account.
+
+    let stake_pda = create_stake(&mut context, &validator, &vote, &config).await;
+
+    let account = get_account!(context, stake_pda);
+    let mut stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set the amount to 100
+    stake_account.amount = 100;
+
+    let updated_stake = Account {
+        lamports: account.lamports,
+        data: stake_account.try_to_vec().unwrap(),
+        owner: account.owner,
+        executable: account.executable,
+        rent_epoch: account.rent_epoch,
+    };
+
+    context.set_account(&stake_pda, &updated_stake.into());
+
+    // When we try to deactivate with an invalid amount (0).
+
+    let deactivate_ix = DeactivateStakeBuilder::new()
+        .config(config)
+        .stake(stake_pda)
+        .stake_authority(authority.pubkey())
+        .amount(0)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[deactivate_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &authority],
+        context.last_blockhash,
+    );
+    let err = context
+        .banks_client
+        .process_transaction(tx)
+        .await
+        .unwrap_err();
+
+    // Then we expect an error.
+
+    assert_custom_error!(err, StakeError::InvalidAmount);
+}
+
+#[tokio::test]
+async fn fail_deactivate_stake_with_uninitialized_stake_account() {
+    let mut context = ProgramTest::new("stake_program", paladin_stake::ID, None)
+        .start_with_context()
+        .await;
+
+    // Given a config account and a validator's vote account.
+
+    let config = create_config(&mut context).await;
```

Comment:
It's not clear in this test exactly _how_ you're breaching the maximum deactivation proportion. The value is hard-coded to 500 basis points over in the setup. Might be nice to (re)configure this value manually over here, or at least add a comment for it?

Reply:
Good point, added a helper that takes the values as arguments and added a comment.